src/main/java/org/simulator/fba/COBRAsolver.java

```diff
@@ -246,10 +247,13 @@ public COBRAsolver(SBMLDocument doc) throws SBMLException, ModelOverdeterminedEx
 			} else {
 				for (Pair<String, Double> pair : species2Reaction.get(species.getId())) {
 					weights[reaction2Index.get(pair.getKey())] = pair.getValue();
+					numberOfReactions++;
```

Comment:
The number of reactions in which a species participates can be obtained without the need to add another counter. You can just get it from `species2Reaction.get(species.getId()).size()`.

Reply:
@draeger, Updated in the next commit